crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -472,15 +472,28 @@ fn stub_file_empty_lines(
         || !stub_suite_can_omit_empty_line(preceding, following, f);
     match kind {
         SuiteKind::TopLevel => {
-            if empty_line_condition {
+            if empty_line_condition
+                || (is_blank_line_after_nested_stub_class_enabled(f.context())
+                    && blank_line_after_nested_stub_class_condition(
+                        preceding.into(),
+                        Some(following.into()),
+                        f,
+                    ))
+            {
```

Comment:
Moving the `is_blank_line_after_nested_stub_class_enabled` into `blank_lines_after_nested_stub_class_condition` simplifies the calling code a lot

Reply:
Yeah that helps. It might still be worth to move the check into the "condition" function because the check is necessary at all call-sites of that function (and leads to very long conditions)